lucene/backward-codecs/src/test/org/apache/lucene/codecs/lucene70/Lucene70RWSegmentInfoFormat.java

```diff
@@ -89,7 +89,10 @@ public void write(Directory dir, SegmentInfo si, IOContext ioContext) throws IOE
       int numSortFields = indexSort == null ? 0 : indexSort.getSort().length;
       output.writeVInt(numSortFields);
       for (int i = 0; i < numSortFields; ++i) {
-        SortField sortField = indexSort.getSort()[i];
+        if (indexSort.getSort()[i] instanceof SortField == false) {
```

Comment:
Hmm, maybe `indexSort` should just continue to take `SortField` for now?  Else this is sort of trappy -- only when `IndexWriter` goes to flush a segment to disk, will we (late) hit this exception?

Or, maybe we could do this check instead in `IndexWriterConfig`?

Reply:
This is test code for a deprecated segment info format, so users should never hit this in production - might be better as an assertion?

Reply:
Woops, sorry, I missed that this was in `backwards-codecs` for test only, you're right.  Let's leave it as a real if?